src/c_api/inspect.rs

```diff
@@ -299,3 +309,118 @@ pub unsafe extern "C" fn blaze_inspector_free(inspector: *mut Inspector) {
+                for (j, sym) in list.into_iter().enumerate() {
+                    let c_sym = unsafe { &(*(*ptr.add(i)).add(j)) };
+                    assert_eq!(
+                        unsafe { CStr::from_ptr(c_sym.name) }.to_bytes(),
+                        CString::new(sym.name).unwrap().to_bytes()
+                    );
+                    assert_eq!(c_sym.address, sym.address);
+                    assert_eq!(c_sym.size, sym.size);
+                    assert_eq!(c_sym.sym_type, blaze_sym_type::from(sym.sym_type));
+                    assert_eq!(c_sym.file_offset, sym.file_offset);
+                    assert_eq!(
+                        unsafe { CStr::from_ptr(c_sym.obj_file_name) }.to_bytes(),
+                        CString::new(
+                            sym.obj_file_name
+                                .as_deref()
+                                .unwrap()
+                                .as_os_str()
+                                .to_os_string()
+                                .into_vec()
+                        )
+                        .unwrap()
+                        .to_bytes()
+                    );
+                }
+            }
+
+            let () = unsafe { blaze_inspect_syms_free(ptr) };
+        }
+
+        let syms = vec![];
+        test(syms);
+
+        let syms = vec![vec![SymInfo {
+            name: "sym1".to_string(),
+            address: 0xdeadbeef,
+            size: 42,
+            sym_type: SymType::Function,
+            file_offset: 1337,
+            obj_file_name: Some(PathBuf::from("/tmp/foobar.so")),
+        }]];
+        test(syms);
+
+        let syms = vec![vec![
+            SymInfo {
+                name: "sym1".to_string(),
+                address: 0xdeadbeef,
+                size: 42,
+                sym_type: SymType::Function,
+                file_offset: 1337,
+                obj_file_name: Some(PathBuf::from("/tmp/foobar.so")),
+            },
+            SymInfo {
+                name: "sym2".to_string(),
+                address: 0xdeadbeef + 52,
+                size: 45,
+                sym_type: SymType::Unknown,
+                file_offset: 1338,
+                obj_file_name: Some(PathBuf::from("other.so")),
+            },
+        ]];
+        test(syms);
+
+        let syms = vec![
+            vec![SymInfo {
+                name: "sym1".to_string(),
+                address: 0xdeadbeef,
+                size: 42,
+                sym_type: SymType::Function,
+                file_offset: 1337,
+                obj_file_name: Some(PathBuf::from("/tmp/foobar.so")),
+            }],
+            vec![SymInfo {
+                name: "sym2".to_string(),
+                address: 0xdeadbeef + 52,
+                size: 45,
+                sym_type: SymType::Unknown,
+                file_offset: 1338,
+                obj_file_name: Some(PathBuf::from("other.so")),
+            }],
+        ];
+        test(syms);
```

Comment:
goodness, it was hard to spot the difference between this test case and the previous one :) maybe a small comment to not scratch head later on?

Reply:
Yeah, it makes sense to comment some more. Will add something.